main.go

```diff
@@ -98,18 +171,181 @@ func main() {
+			} else {
+				postlang = ch.Language
+			}
 			p := &writeas.PostParams{
-				Title:      wpp.Title,
-				Slug:       wpp.PostName,
-				Content:    con,
-				Created:    &wpp.PostDateGmt,
-				Updated:    &wpp.PostDateGmt,
-				Font:       "norm",
-				Language:   &ch.Language,
-				Collection: coll.Alias,
+				Title:    wpp.Title,
+				Slug:     wpp.PostName,
+				Content:  con,
+				Created:  &wpp.PostDateGmt,
+				Updated:  &wpp.PostDateGmt,
+				Font:     "norm",
+				Language: &postlang,
 			}
+			c.Posts = append(c.Posts, p)
+		}
+		coll.Collections = append(coll.Collections, c)
+	}
+	return coll
+}
+
+// Temporarily using an ini file to store instance tokens.
+// This is probably not what the rest of the code does,
+// but I need some way to handle this for now.
+// TODO: Get this in line with the rest of the code (see T586)
+
+// ini file format:
+// Each instance has its own [section]
+// Semicolons (;) at the beginning of a line indicate a comment
+// Can't start a comment mid-line (this allows semicolons in variable values)
+// Blank lines are ignored
+func importConfig() map[string]instance {
+	file, err := ioutil.ReadFile("instances.ini")
+	if err != nil {
+		errQuit("Error reading instances.ini")
+	}
+	lines := strings.Split(string(file), "\n")
+	instances := make(map[string]instance)
+	curinst := ""
+	newinst := instance{}
+	for i := 0; i < len(lines); i++ {
+		line := lines[i]
+		if line == "" {
+			continue
+		}
+		fc := string(line[0])
+		if fc == ";" {
+			continue
+		}
+		if fc == "[" {
+			if curinst != "" {
+				instances[curinst] = newinst
+				newinst = instance{}
+			}
+			curinst = line[1:(len(line) - 1)]
+		} else {
+			loc := strings.Index(line, "=")
+			if curinst == "" || loc == -1 {
+				errQuit("Malformed ini file")
+			}
+			k := line[:loc]
+			v := line[loc+1:]
+			if k == "url" {
+				newinst.Url = v
+			} else if k == "token" {
+				newinst.Token = v
+			} else {
+				errQuit("Malformed ini file")
+			}
+		}
+	}
+	instances[curinst] = newinst
+	return instances
+}
+
+func main() {
+	a := parseArgs(os.Args)
+	// if len(os.Args) < 2 {
+	// 	//errQuit("usage: wp-import https://write.as filename.xml")
+	// 	errQuit("usage: wp-import filename.xml")
+	// }
+	// fname := os.Args[1]
+	fname := a["filename"]
+	inst := "writeas"
+	if a["instance"] != "" {
+		inst = a["instance"]
+	}
+
+	instances := importConfig()
+	//fmt.Println(instances)
+	var cl *writeas.Client
+	t := ""
+	u := ""
+	if val, ok := instances[inst]; ok {
+		t = val.Token
+		u = val.Url
+		cl = writeas.NewClientWith(writeas.Config{
+			URL:   u + "/api",
+			Token: t,
+		})
+	} else {
+		fmt.Println("We don't have a token for " + inst + ".")
+		r := bufio.NewScanner(os.Stdin)
+		fmt.Print("Instance URL: ")
+		r.Scan()
+		url := r.Text()
+		if string(url[:5]) != "https" {
+			url = "https://" + url
+		}
+		if string(url[len(url)-1:]) == "/" {
+			url = string(url[:len(url)-1])
+		}
+		//fmt.Println("Using URL", url)
```

Comment:
this and the next two commented `Println`s, are they meant to be printed or were they for debugging?

Reply:
Debugging, yes.